macOS/DuckDuckGo/VisualRefresh/ThemeColors.swift

```diff
@@ -0,0 +1,167 @@
+        destructiveContentSecondary = NSColor(designSystemColor: .destructiveContentSecondary, palette: palette)
+        destructiveContentTertiary = NSColor(designSystemColor: .destructiveContentTertiary, palette: palette)
+        destructiveGlow = NSColor(designSystemColor: .destructiveGlow, palette: palette)
+        destructivePrimary = NSColor(designSystemColor: .destructivePrimary, palette: palette)
+        destructiveSecondary = NSColor(designSystemColor: .destructiveSecondary, palette: palette)
+        destructiveTertiary = NSColor(designSystemColor: .destructiveTertiary, palette: palette)
+        destructiveTextPrimary = NSColor(designSystemColor: .destructiveTextPrimary, palette: palette)
+        destructiveTextSecondary = NSColor(designSystemColor: .destructiveTextSecondary, palette: palette)
+        destructiveTextTertiary = NSColor(designSystemColor: .destructiveTextTertiary, palette: palette)
+        highlightPrimary = NSColor(designSystemColor: .highlightPrimary, palette: palette)
+        iconsPrimary = NSColor(designSystemColor: .iconsPrimary, palette: palette)
+        iconsSecondary = NSColor(designSystemColor: .iconsSecondary, palette: palette)
+        iconsTertiary = NSColor(designSystemColor: .iconsTertiary, palette: palette)
+        shadowPrimary = NSColor(designSystemColor: .shadowPrimary, palette: palette)
+        shadowSecondary = NSColor(designSystemColor: .shadowSecondary, palette: palette)
+        shadowTertiary = NSColor(designSystemColor: .shadowTertiary, palette: palette)
+        surfaceBackdrop = NSColor(designSystemColor: .surfaceBackdrop, palette: palette)
+        surfaceCanvas = NSColor(designSystemColor: .surfaceCanvas, palette: palette)
+        surfacePrimary = NSColor(designSystemColor: .surfacePrimary, palette: palette)
+        surfaceSecondary = NSColor(designSystemColor: .surfaceSecondary, palette: palette)
+        surfaceTertiary = NSColor(designSystemColor: .surfaceTertiary, palette: palette)
+        textPrimary = NSColor(designSystemColor: .textPrimary, palette: palette)
+        textSecondary = NSColor(designSystemColor: .textSecondary, palette: palette)
+        textTertiary = NSColor(designSystemColor: .textTertiary, palette: palette)
+        toneShadePrimary = NSColor(designSystemColor: .toneShadePrimary, palette: palette)
+        toneTintPrimary = NSColor(designSystemColor: .toneTintPrimary, palette: palette)
+    }
+
+    private static func designColorPalette(for themeName: ThemeName) -> DesignResourcesKit.ColorPalette {
+        switch themeName {
+        case .default:
+            .default
+        case .figma:
+            .figma
+        case .coolGray:
+            .coolGray
+        case .desert:
+            .desert
+        case .green:
+            .green
+        case .orange:
+            .orange
+        case .rose:
+            .rose
+        case .slateBlue:
+            .slateBlue
+        case .violet:
+            .violet
+        }
+    }
+}
```

Comment:
This class is essentially an adapter between `DesignResourcesKit.ColorPalette` and `ColorPalette`.

Current plan is to rename `ColorPalette` (from the Mac App’s namespace) into `ThemeColorsProtocol` or `ThemeColorsProvider`.

Alternatively, we can explore changing the API in the `DesignResourcesKit` package. My initial iteration was meant to avoid the need for an adapter, but that shifted, due to feedback received during review.

@jotaemepereira Would very much love to hear your thoughts, thanks in advance!